File: pueblo_golf_tournament_mobile/lib/dto/model/payment-channel-account-dto.dart
import 'package:pueblo_golf_tournament_mobile/base/base-dto.dart';

import 'package:json_annotation/json_annotation.dart';
import 'package:pueblo_golf_tournament_mobile/dto/model/payment-channel-dto.dart';

part 'payment-channel-account-dto.g.dart';

@JsonSerializable()
class PaymentChannelAccountDto extends BaseDto {
  final int paymentChannelId;
  final int tournamentId;
  final String accountName;
  final String accountNumber;
  final PaymentChannelDto? paymentChannel;
  PaymentChannelAccountDto({
    required super.id,
    required this.paymentChannelId,
    required this.tournamentId,
    required this.accountName,
    required this.accountNumber,
    this.paymentChannel,
  });

  factory PaymentChannelAccountDto.fromJson(Map<String, dynamic> json) =>
      _$PaymentChannelAccountDtoFromJson(json);
  Map<String, dynamic> toJson() => _$PaymentChannelAccountDtoToJson(this);
}

File: pueblo-golf-tournament-api/Dto/PaymentChannelAccountDto.cs
using pueblo_golf_tournament_api.Base;
using pueblo_golf_tournament_api.Entities;

namespace pueblo_golf_tournament_api.Dto
{
    public class PaymentChannelAccountDto : BaseDto
    {
        public long PaymentChannelId { get; set; }
        public long? TournamentId { get; set; }
        public string? AccountName { get; set; }
        public string? AccountNumber { get; set; }
        public virtual PaymentChannelDto? PaymentChannel { get; set; }
    }
}